File: mathematica/fun/test_string.m
(* test_string.m
                               wookay.noh at gmail.com
*)

assertEqual[expected_, got_] :=
  WriteString["stdout",
    If[expected == got,
      StringJoin["passed: ", ToString[expected], "\n"],
      StringJoin["Assertion failed\nExpected: ", ToString[expected],
        "\nGot: ", ToString[got], "\n"]]]


assertEqual[ {"a", "b", "c"} , StringSplit["a,b,c", ","]                      ]
assertEqual[ "abc"           , StringJoin[{"a", "b", "c"}]                    ]
assertEqual[ "abcdefg"       , StringJoin[CharacterRange["a", "g"]]           ]
assertEqual[ {1, Plus, 2}    , Riffle[{1, 2}, Plus]                           ]
assertEqual[ "1+2+3"         , StringJoin[Riffle[Map[ToString,{1,2,3}], "+"]] ]
assertEqual[ "a,b,c"         , StringJoin[Riffle[{"a","b","c"}, ","]]         ]
assertEqual[ "abc,"          , StringJoin[{"a","b","c"}, ","]                 ]
assertEqual[ "abc"           , StringTrim[" abc "]                            ]
assertEqual[ "abc"           , StringReplace["a,b,c", "," -> ""]              ]

assertEqual[ "cba"           , StringReverse["abc"]                           ]
assertEqual[ 3               , StringLength["abc"]                            ]

assertEqual[ {"ㄱ", "ㄲ", "ㄳ", "ㄴ", "ㄵ", "ㄶ", "ㄷ", "ㄸ", "ㄹ", "ㄺ", "ㄻ", "ㄼ", "ㄽ", "ㄾ", "ㄿ", "ㅀ", "ㅁ", "ㅂ", "ㅃ", "ㅄ", "ㅅ", "ㅆ", "ㅇ", "ㅈ", "ㅉ", "ㅊ", "ㅋ", "ㅌ", "ㅍ", "ㅎ"} ,
                               CharacterRange["ㄱ", "ㅎ"]                     ]

Exit[]

File: perl/test_string.pl
# test_string.pl
#                           wookay.noh at gmail.com

sub assert_equal {
  my $expected = $_[0];
  my $got = $_[1];
  if ($expected eq $got) {
    print "passed: $expected\n";
  } else {
    print "Assertion failed\nExpected: $expected\nGot: $got\n";
  }
}


assert_equal( "apple" , lc("APPLE") );
assert_equal( "aPPLE" , lcfirst("APPLE") );
assert_equal( "APPLE" , uc("apple") );
assert_equal( "Apple" , ucfirst("apple") );
assert_equal( 5, length("apple") );
assert_equal( 1 , index("apple", "p") );
assert_equal( 2 , rindex("apple", "p") );

$str = "apple";
assert_equal( "e" , chop($str) );
assert_equal( "appl" , $str );
assert_equal( 0 , chomp($str) );

$str = "apple\n";
assert_equal( 1 , chomp($str) );
assert_equal( "apple" , $str );

$str = "1 + 2";
assert_equal( 3 , eval($str) );
